activerecord/test/cases/callbacks_test.rb

```diff
@@ -78,6 +78,42 @@ def cancel
     end
 end
 
+class RelatedProject < ActiveRecord::Base
+  self.table_name = 'projects'
+end
+
+class RelatedDeveloper < ActiveRecord::Base
+  self.table_name = 'developers'
+end
+
+class RelatedDeveloperProject < ActiveRecord::Base
+  self.table_name = 'developers_projects'
+
+  belongs_to :developer, :class => RelatedProject, :foreign_key => 'developer_id'
```

Comment:
Since no backwards compatibility is planned for this, we can use Ruby >= 1.9 syntax for hashes